app/controllers/katello/api/v2/content_uploads_controller.rb

```diff
@@ -9,8 +9,12 @@ class Api::V2::ContentUploadsController < Api::V2::ApiController
     api :POST, "/repositories/:repository_id/content_uploads", N_("Create an upload request")
     param :repository_id, :number, :required => true, :desc => N_("repository id")
     param :size, :number, :required => true, :desc => N_("Size of file to upload")
+    param :checksum, String, :required => true, :desc => N_("Checksum of file to upload")
```

Comment:
since this param is not used for pulp2, can we make it optional (not required?).  I'd prefer to not break api compatibility until we absolutely have to.  Maybe have a check in the pulp3 content service that raises an exception if its not passed in for pulp3 types.

Reply:
So in hammer, regardless of pulp2/pulp3, we are passing checksum to keep the hammer part consistent. So this check, while useless for pulp2 is satisfied from hammer. It is easier and cleaner this way. What do you think?

Reply:
right, i'm not worried as much about hammer, as i am about other api users.  Which it probably still works if you're not using pulp2 right?  so in that sense it is optional under certain situations?

Can you make it optional and just always pass it with hammer?  That way individual api users won't be super confused

Reply:
I opened this https://projects.theforeman.org/issues/28106 to track making it required.